downloaded/authorizer.openapi.json

```diff
@@ -1,2645 +0,0 @@
-{
```

Comment:
the `downloaded` folder is intended to be checked in to make builds reproducable. downloads has inputs, generated has outputs.

Reply:
Maybe there's something I'm missing about the workflow. As I understand it:
- the consumer of the `downloaded` folder is the `generate.sh` script
- the generate script removes any existing `*.openapi.json` files and refetches them from the `openapi-grpc` repo

So to me they seem the files in this folder are like intermediate build artifacts, but we don't actually benefit from any sort of caching since they always get cleaned up when running the script. They seem like at best they could be used for debugging when the generated TS definitions aren't as expected.

Reply:
I agree that the commit hash itself is sufficient. However, I chose to also check in these intermediary artifacts because:
 - You can re-run any build locally without having to set Environment Variables, making development much less tedious.
 - Diffs of this repo are now self-contained. You can scan through the git commit history and get the full picture. This will be useful in eliminating a variable when we deal with issues in new versions of the TS generator.
 - Calls to generate are not bound to external network calls. Any non-determinism is left to the github actions; the repo builds like any other source code repo

Both approaches, in the end, produce the same output artifacts. I feel that storing the downloaded content does meaningfully improve the developer experience and do not impose any meaningful cost.

Reply:
> You can re-run any build locally without having to set Environment Variables, making development much less tedious.
> Calls to generate are not bound to external network calls. Any non-determinism is left to the github actions; the repo builds like any other source code repo

To make these points actually a reality, can we add a flag to the generate.sh script that makes it not make network requests by default? Otherwise, one has to comment out the `rm` and `curl` commands from the script when running locally.
`$ ./generate.sh  # Uses whatever is already in downloaded/`
`$ ./generate.sh --fetch-specs  # Remove whatever is in downloaded/ and refetch from openapi-grpc repo`

> Diffs of this repo are now self-contained. You can scan through the git commit history and get the full picture

This assumes that the files in `downloaded` are actually the ones used to generate the type definitions. This will be inherently true when this is run by the Github workflow, but there's no automated process stopping me from making local changes to the `downloaded` files without regenerating the type definitions. I don't really have an actionable point here, just saying this requires some level of trust.

Reply:
I think the flag makes sense, sure. Feel free to add it, or let me know and I'll PR it.

Agreed on the trust point. I think we should strive for this repo to become a mechanical projection of its upstream repo. Manual changes should be rare and highly scrutinized.

Reply:
I added the flag, albeit in a naive way.